workflow/controller/operator.go

```diff
@@ -258,18 +312,26 @@ func (woc *wfOperationCtx) podReconciliation() error {
 	}
 	seenPods := make(map[string]bool)
 	for _, pod := range podList.Items {
-		seenPods[pod.ObjectMeta.Name] = true
-		if node, ok := woc.wf.Status.Nodes[pod.ObjectMeta.Name]; ok {
+		nodeNameForPod := pod.Annotations[common.AnnotationKeyNodeName]
+		nodeID := woc.wf.NodeID(nodeNameForPod)
+		seenPods[nodeID] = true
+		if node, ok := woc.wf.Status.Nodes[nodeID]; ok {
 			if newState := assessNodeStatus(&pod, &node); newState != nil {
-				woc.wf.Status.Nodes[pod.ObjectMeta.Name] = *newState
+				woc.wf.Status.Nodes[nodeID] = *newState
 				woc.updated = true
 			}
 			if woc.wf.Status.Nodes[pod.ObjectMeta.Name].Completed() {
 				woc.completedPods[pod.ObjectMeta.Name] = true
 			}
 		}
 	}
-	if len(podList.Items) > 0 {
+
+	retriesLeft, err := woc.processNodesWithRetries()
+	if err != nil {
+		return err
+	}
+
+	if retriesLeft || len(podList.Items) > 0 {
```

Comment:
Why do we need to consult retriesLeft to see if we should continue the check for deleted pods? Shouldn't this be left as:
```
if len(podList.Items) > 0 {
```

Reply:
Thanks for pointing this out again. At some point, there lower `assessNodeStatus` was marking the intermediate node as failed with a `pod deleted` error and therefore I had to bypass that. No longer needed.

`processNodesWithRetries` only returns an error now.